### ISSUE TITLE
Missing documentation about type=branch on workflow_run events

### ISSUE THREAD

**Issue by overbit** [docker/metadata-action#259]
#### Expected behaviour

A mention of the default behaviour of type=`branch` is missing and it can lead users to unexpected results. 

#### Actual behaviour

Updating the documentation with the default behaviour for events that are not listed.

### Configuration

* Repository URL (if public): N/A
* Build URL (if public): N/A

```yml
# paste your YAML workflow file here and remove sensitive data
N/A
```

### Logs

> Download the [log file of your build](https://docs.github.com/en/actions/managing-workflow-runs/using-workflow-run-logs#downloading-logs)
> and [attach it](https://docs.github.com/en/github/managing-your-work-on-github/file-attachments-on-issues-and-pull-requests) to this issue.

**Comment by crazy-max** [comment]
There is no `type=branch`, did you mean `type=ref,event=branch`?

**Comment by overbit** [comment]
Yes @crazy-max, you are right. I'm updating the issue description.

**Comment by overbit** [comment]
My current expectation from the documentation is that `type=ref,event=branch` will only be enabled for workflows triggered by events related to the branch, such as `push`, `workflow_dispatch`, `release`, and `workflow_run`.
But for `workflow_run` I did expect to use the triggering workflow branch instead of `github.ref`.

What do you thing @crazy-max ?

**Comment by crazy-max** [comment]
The action does not support the [`workflow_run` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run). I will look at it.

**Comment by polarathene** [comment]
> But for `workflow_run` I did expect to use the triggering workflow branch instead of `github.ref`.

That's how that trigger always worked? 

[`workflow_run` event docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run):

> **Note:** This event will only trigger a workflow run **if** the workflow file is on the **default branch**.

So you should expect to only configure these from that default branch? How are you trying to leverage `workflow_run` with this action? Pull requests?

---

Unrelated to this action, I have used `workflow_run` for PR preview builds of docs where you need to volley an untrusted build over to a trusted context (_where you don't want to run third-party code in_):

- [Prepare preview](https://github.com/docker-mailserver/docker-mailserver/blob/82c38f242648290ab109de8e3994ea7564120deb/.github/workflows/docs-preview-prepare.yml) workflow (_untrusted context, runs in PR branch_)
- [Deploy preview](https://github.com/docker-mailserver/docker-mailserver/blob/82c38f242648290ab109de8e3994ea7564120deb/.github/workflows/docs-preview-deploy.yml) workflow (_trusted context via `workflow_run`, runs from default branch_)

As you can see in the deploy workflow, there is limited context of the original workflow branch (PR), some of the context needed is volleyed over via export + import of ENV, while other context from `workflow_run` can be used:
- `github.event.workflow_run.event == 'pull_request'`
- `github.event.workflow_run.head_sha`
- `github.event.workflow_run.id`
- `github.event.workflow_run.conclusion == 'success'`

**Comment by feryardiant** [comment]
Any plan on this?
